File: fronted/app/src/main/java/com/asflum/cashewregister/BackendProcessor.kt
package com.asflum.cashewregister

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import okhttp3.Response
import okio.IOException

object BackendProcessor {

    private const val API_URL = BuildConfig.API_URL
    private val client = OkHttpClient()

    suspend fun processExpenses(rawExpenses: String): Result<String> {
        return try {
            val requestBody = rawExpenses.toRequestBody("application/json".toMediaType())

            val request = Request.Builder()
                .url("${API_URL}/process-expenses")
                .post(requestBody)
                .build()

            val response: Response = withContext(Dispatchers.IO) {
                client.newCall(request).execute()
            }

            Result.success(response.body.toString())
        } catch (_: IOException) {
            Result.failure(Exception("Error de red. Revise su conexión e intente de nuevo."))
        } catch (e: Exception) {
            Result.failure(Exception("Error inesperado: ${e.localizedMessage}"))
        }
    }
}

File: frontend/lib/services/backend_processor.dart
import 'package:expense_tracker/utils/result.dart';
import 'dart:convert';
import 'package:http/http.dart' as http;

class BackendProcessor {
  Future<Result<String>> processMessages(List<Map<String, dynamic>> messages) async {
    try {
      final String apiUrl = const String.fromEnvironment('API_URL');
      final body = jsonEncode(messages);

      final url = Uri.parse('$apiUrl/process-expenses');

      final response = await http.post(
        url,
        headers: {'Content-Type': 'application/json'},
        body: body,
      );

      if (response.statusCode == 200) {
        return Result.success(response.body);
      } else {
        return Result.failure(
          Exception("Error del servidor: ${response.body}"),
        );
      }
    } catch (e) {
      return Result.failure(Exception("Ocurrio un error inesperado: $e"));
    }
  }
}